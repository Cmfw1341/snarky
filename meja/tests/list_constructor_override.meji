type nil;  type t('a) = [] : t(nil) | (::)('hd, t('tl)) : t('hd -> 'tl); 
let x : t(nil);  let y : t(int -> nil); 
let z : t(int -> bool -> unit -> nil); 
let z : t(int -> bool -> unit -> nil); 
module A : { type u('a, 'b) = []('a, 'b);  let x : u(int, bool);  }; 

