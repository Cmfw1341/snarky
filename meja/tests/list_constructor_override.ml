open Snarky
open Snarky.Snark
module Impl =
  Snarky.Snark.Run.Make (Snarky.Backends.Mnt4.Default) (Core_kernel.Unit)
open Impl

type nil

type 'a t = [] : nil t | ( :: ) : 'hd * 'tl t -> ('hd -> 'tl) t

let x = []

let y = [Field.constant (Field.Constant.of_string "12")]

let z = [Field.constant (Field.Constant.of_string "1"); true; ()]

let z = [Field.constant (Field.Constant.of_string "1"); true; ()]

module A = struct
  type ('a, 'b) u = [] of 'a * 'b

  let x = []
end
