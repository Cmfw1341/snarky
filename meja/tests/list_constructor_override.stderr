File "tests/list_constructor_override.meja", line 9, characters 13-17:
Error: Unbound constructor true.
